lib/chutney/TorNet.py

```diff
@@ -1180,7 +1214,11 @@ def isBootstrapped(self):
         """Return true iff the logfile says that this instance is
            bootstrapped."""
         pct, _, _ = self.getLastBootstrapStatus()
-        return pct == LocalNodeController.SUCCESS_CODE
+        if self.getOnionService():
```

Comment:
I think this logic is the wrong way around?

If it's not an onion service, we just want to check the standard `getLastBootstrapStatus()`.

If it is an onion service, we want to check *both* `getLastBootstrapStatus()` and `getLastOnionServiceDescStatus()`. We should only call an onion service bootstrapped if both checks pass.

Reply:
I also wonder why the tests pass with the logic this way around?

Reply:
Oh, I see. `isBootstrapped()` is an unused function. That's really confusing. We should use it, or delete it.

The actual code that checks for bootstrap is here:
https://github.com/torproject/chutney/blob/master/lib/chutney/TorNet.py#L2244

Reply:
I vote for refactoring so that the function gets used. I'll look into it.